A66-otel-stats.md

```diff
@@ -0,0 +1,542 @@
+    0.8, 1, 2, 5, 10, 20, 50, 100
+*   Size : 0, 1024, 2048, 4096, 16384, 65536, 262144, 1048576, 4194304,
+    16777216, 67108864, 268435456, 1073741824, 4294967296
+*   Count : 0, 1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096, 8192,
+    16384, 32768, 65536
+
+These buckets were chosen to maintain compatibility with the gRPC OpenCensus
+spec. The OpenTelemetry API has added an experimental feature for
+[advice](https://opentelemetry.io/docs/specs/otel/metrics/api/#instrument-advice)
+that would allow the gRPC library to provide these buckets as a hint. Since this
+is still an experimental feature and not yet implemented in all languages, it is
+upto the user to choose the right bucket boundaries.
+
+Also note that, as per an
+[OpenTelemetry proposal on stability](https://docs.google.com/document/d/1Nvcf1wio7nDUVcrXxVUN_f8MNmcs0OzVAZLvlth1lYY/edit#heading=h.dy1cg9doaq26)
+though, changes to bucket boundaries might not be considered a breaking change.
+Depending on the proposal, this recommendation would change to use
+`ExponentialHistogram`s instead, which would allow for automatic adjustments of
+the scale to better fit the data.
+
+#### Attributes
+
+*   `grpc.method` : Full gRPC method name, including package, service and
+    method, e.g. "google.bigtable.v2.Bigtable/CheckAndMutateRow". Note that some
+    gRPC implementations allow server to handle generic method names, i.e., not
+    registering method names in advance with the server. This allows clients to
+    send arbitrary method names that could potentially open up the server to
+    malicious attacks that result in metrics being stored with a high
+    cardinality. To prevent this, unregistered/generic method names should by
+    default be reported with "other" value instead. Implementations should
+    provide the option to override this behavior to allow recording generic
+    method names as well.
+*   `grpc.status` : gRPC server status code received, e.g. "OK", "CANCELLED",
+    "DEADLINE_EXCEEDED".
+    [(Full list)](https://grpc.github.io/grpc/core/md_doc_statuscodes.html)
+*   `grpc.target` : Canonicalized target URI used when creating gRPC Channel,
+    e.g. "dns:///pubsub.googleapis.com:443", "xds:///helloworld-gke:8000".
+    Canonicalized target URI is its form with the scheme if the user didn't
+    mention the scheme. For channels such as inprocess channels where a target
+    URI is not available, implementations can synthesize a target URI. It is
+    possible for some channels to use IP addresses as target strings and this
+    might again blow up the cardinality. Implementations should provide the
+    option to override recorded target names with "other".
+
+#### Client Per-Attempt Instruments
+
+*   **grpc.client.attempt.started** <br>
+    The total number of RPC attempts started, including those that have not completed. <br>
+    *Attributes*: grpc.method, grpc.target <br>
+    *Type*: Counter <br>
+    *Unit*: {attempt} <br>
```

Comment:
Is `{attempt}` supposed to be recorded as OpenTelemetry metric [unit](https://opentelemetry.io/docs/specs/otel/metrics/semantic_conventions/#units) as is?

Reply:
Yes, as per https://opentelemetry.io/docs/specs/otel/metrics/semantic_conventions/

Reply:
Can we also add unit and description for this as well other metrics?

Reply:
Following the recommendations at https://opentelemetry.io/docs/specs/otel/metrics/semantic_conventions/#instrument-units

Reply:
Added the unit. I don't think description matters so much but for consistency, we can use the description at https://github.com/grpc/grpc/pull/34441